cms/templates/visibility_editor.html

```diff
@@ -4,25 +4,42 @@
 from django.utils.translation import ugettext as _
 from contentstore.utils import ancestor_has_staff_lock, get_visibility_partition_info
 from openedx.core.djangolib.markup import HTML, Text
+from lms.lib.utils import is_unit
 
 partition_info = get_visibility_partition_info(xblock)
 selectable_partitions = partition_info["selectable_partitions"]
 selected_partition_index = partition_info["selected_partition_index"]
 selected_groups_label = partition_info["selected_groups_label"]
 
 is_staff_locked = ancestor_has_staff_lock(xblock)
+block_is_unit = is_unit(xblock)
+block_type = "unit" if block_is_unit else "component"
```

Comment:
This won't work because "unit" and "component" will not get translated. So the rest of the sentence would get translated, and one word would be left in English.

Unfortunately you will need to duplicate the whole strings based on the value of "block_is_unit".